sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3718,6 +3718,20 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-33677: LikeSimplification should be skipped if pattern contains any escapeChar") {
+    withTable("string_tbl") {
+      sql("CREATE TABLE string_tbl USING parquet SELECT 'm@ca' AS s")
```

Comment:
nit: could you use a temporary table for better test performance?

Reply:
there is no temporary table in spark, you mean temporary view?

Reply:
Unfortunately `Like` will be constant folded and not passed to `LikeSimplification` If using temp view here.

Reply:
Oh, yea, I meant a temporary view...

Reply:
@luluorta have you tried temp view? AFAIK `ConvertToLocalRelation` is disabled in tests.

Reply:
It works if I use DataFrame to create temp view.